app/models/ems_refresh/save_inventory_infra.rb

```diff
@@ -306,4 +290,42 @@ def save_storage_files_inventory(storage, hashes)
     deletes = storage.storage_files(true).dup
     save_inventory_multi(:storage_files, storage, hashes, deletes, [:name])
   end
+
+  def find_host(h, ems_id)
+    if h[:ems_ref]
+      _log.debug "EMS ID: #{ems_id} Host database lookup - ems_ref: [#{h[:ems_ref]}] ems_id: [#{ems_id}]"
+      found = Host.find_by_ems_ref_and_ems_id(h[:ems_ref], ems_id)
+    end
+
+    if found.nil?
+      if h[:hostname].nil? && h[:ipaddress].nil?
+        _log.debug "EMS ID: #{ems_id} Host database lookup - name [#{h[:name]}]"
+        found = Host.where(:ems_id => ems_id).detect { |e| e.name.downcase == h[:name].downcase }
+      elsif ["localhost", "localhost.localdomain", "127.0.0.1"].include_none?(h[:hostname], h[:ipaddress])
+        # host = Host.find_by_hostname(hostname) has a risk of creating duplicate hosts
+        # allow a deleted EMS to be re-added an pick up old orphaned hosts
+        _log.debug "EMS ID: #{ems_id} Host database lookup - hostname: [#{h[:hostname]}] IP: [#{h[:ipaddress]}] ems_ref: [#{h[:ems_ref]}]"
+        found = look_up_host(h[:hostname], h[:ipaddress], :ems_ref => h[:ems_ref])
+      end
+    end
+
+    found
+  end
+
+  def look_up_host(hostname, ipaddr, opts = {})
```

Comment:
`find_orphaned_host` or something? Without reading the method body, it seems hard to imagine the difference between `find_` and `look_up_`.